src/CreateMiddleware/CreateMiddleware.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-tooling for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-tooling/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Tooling\CreateMiddleware;
+
+/**
+ * Create middleware
+ *
+ * Creates a middleware class file for a given class in a given project root.
+ */
+class CreateMiddleware
+{
+    /**
+     * @var string Template for middleware class.
+     */
+    const CLASS_SKELETON = <<< 'EOS'
+<?php
+
+namespace %namespace%;
+
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface;
+use Psr\Http\ServerRequestInterface;
+
+class %class% implements MiddlewareInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function process(ServerRequestInterface $request, DelegateInterface $delegate)
+    {
+        // $response = $delegate->process($request);
+    }
+}
+EOS;
+
+    /**
+     * @param string $class
+     * @param string $projectRoot
+     * @return string
+     * @throws CreateMiddlewareException
```

Comment:
Yes. This is the only public method on an instance, so we need to report any exceptions that could be thrown by invoking it. In this particular case, it delegates to private methods that can throw this exception.